modules/RTC/JitsiLocalTrack.js

```diff
@@ -74,6 +74,13 @@ export default class JitsiLocalTrack extends JitsiTrack {
         // track resolution.
         this.resolution = RTCBrowserType.isFirefox() ? null : resolution;
 
+        if (RTCBrowserType.usesNewGumFlow()) {
+            // For new gum flow, getSettings should be available... Get the
+            // resolution from the track itself because it cannot be certain
+            // which resolution webrtc has fallen back to using.
+            this.resolution = track.getSettings().height;
```

Comment:
shouldn't this be the smallest of height or width? (I'm thinking a portrait orientation)

Reply:
is this what the consumers of this field expect?  if so we could change that.  though right now this flow only applies to chrome...in general we've tried to defer handling of scenarios introduced by other clients until we move those clients over to the new flow, so maybe we can hold off on changing this until portrait-able devices get moved over.

Reply:
I can put my 24 inch monitor in portrait mode :-)

Reply:
this isn't about display, though, this is about capture

Reply:
One problem here is how the instance variable resolution is being used. Its primary usage is in ConnectionQuality, where it is used to calculate a connection quality value. The width (jitsiLocalTrack.resolution) is used to look up a height in the hardcoded hash Resolutions, which will return an assumed height and width combo. From there calculation are done based on the height and width. ConnectionQuality and Resolutions does not account for portrait. Later on AvgRTPStatsReporter will take the height and width combo and do some average pixel calculation. I haven't looked at what exactly is being calculated at any point in the stats, just how resolution is being used.